File: solutions/dart/difference-of-squares/1/lib/difference_of_squares.dart
import 'dart:math';

class DifferenceOfSquares {

  int _sum_pa(int n) {
    return ((1 + n) * n) ~/ (2);
  }

  int squareOfSum(int n) {

    return pow(_sum_pa(n), 2).toInt();
  }

  int sumOfSquares(int n) {
    Iterable<int> nums() sync* {
      for (int i = 1; i <= n; i++)
        yield pow(i, 2).toInt();
    }
    return nums().fold(0, (a, b) => a + b);
  }

  int differenceOfSquares(int n) {
    return squareOfSum(n) - sumOfSquares(n);
  }
}

File: solutions/ocaml/difference-of-squares/1/difference_of_squares.ml
let sum_series n = (n * (n + 1)) / 2

let square_of_sum n = 
  let x = sum_series n
  in x * x

let sum_of_squares n =
  ((n * n + n) * (2 * n + 1)) / 6

let difference_of_squares n  =
  square_of_sum n - sum_of_squares n 


